tools/generator/src/Generator/XCSchemeInfo+BuildActionInfo.swift

```diff
@@ -54,17 +62,31 @@ extension XCSchemeInfo.BuildActionInfo {
         guard let buildAction = buildAction else {
             return nil
         }
+        let context = "creating a `BuildActionInfo`"
         let buildTargetInfos: [XCSchemeInfo.BuildTargetInfo] = try buildAction.targets
             .map { buildTarget in
                 let targetID = try targetIDsByLabel.value(
                     for: buildTarget.label,
-                    context: "creating a `BuildActionInfo`"
+                    context: context
                 )
                 return XCSchemeInfo.BuildTargetInfo(
                     targetInfo: try targetResolver.targetInfo(targetID: targetID),
                     buildFor: buildTarget.buildFor
                 )
             }
-        try self.init(targets: buildTargetInfos)
+
+        try self.init(
+            targets: buildTargetInfos,
+            preActions: buildAction.preActions.prePostActionInfos(
+                targetResolver: targetResolver,
+                targetIDsByLabel: targetIDsByLabel,
+                context: context
```

Comment:
We should have a unique context for each of these. This one could be `` "creating a pre-action `PrePostActionInfo`" ``.

Reply:
Makes sense, I will add the unique contexts.